### ISSUE TITLE
Problem with genes slot

### ISSUE THREAD

**Issue by federicomarini** [aoles/DEFormats#4]
Hi Andrzej,

quick in the office after the vacation, and putting together the code for the reproducible example of this issue.

It seems DEFormats *expects* the complete infos on the `genes` slot from the DGEList. This means also their GRanges.

```
> library(Glimma)
> library(limma)
> library(edgeR)
> data(lymphomaRNAseq)
> rnaseq <- lymphomaRNAseq
> head(rnaseq$genes)
      GeneID Length Symbols                                            GeneName
11287  11287   4681     Pzp                              pregnancy zone protein
11298  11298   1455   Aanat                  arylalkylamine N-acetyltransferase
11302  11302   5743    Aatk                apoptosis-associated tyrosine kinase
11303  11303  10260   Abca1 ATP-binding cassette, sub-family A (ABC1), member 1
11304  11304   7248   Abca4 ATP-binding cassette, sub-family A (ABC1), member 4
11305  11305   8061   Abca2 ATP-binding cassette, sub-family A (ABC1), member 2
      Chr
11287   6
11298  11
11302  11
11303   4
11304   3
11305   2
> library(DEFormats)
> as.DESeqDataSet(rnaseq)
Error in .find_start_end_cols(df_colnames0, start.field0, end.field0) : 
  cannnot determine start/end columns
> rnaseq$genes <- NULL
> as.DESeqDataSet(rnaseq)
class: DESeqDataSet 
dim: 27179 7 
metadata(1): version
assays(1): counts
rownames(27179): 11287 11298 ... 100862405 100862406
rowData names(0):
colnames(7): smchd1-nv11-1940_AD2E00ACXX_TTAGGC_L004.sam
  smchd1-nv11-2086_AD2E00ACXX_TGACCA_L004.sam ...
  smchd1-wt-241_AD2E00ACXX_ATCACG_L004.sam
  NJ_RNA1__Smchd1_WT_lymphoma_1934__C2G5CACXX_AGTCAA_L003.sam
colData names(5): sampleID genotype group lib.size norm.factors
> sessionInfo()
R version 3.3.0 (2016-05-03)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: OS X 10.10.5 (Yosemite)

locale:
[1] de_DE.UTF-8/de_DE.UTF-8/de_DE.UTF-8/C/de_DE.UTF-8/de_DE.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] DEFormats_1.2.0 edgeR_3.16.5    limma_3.30.11   Glimma_1.2.1   

loaded via a namespace (and not attached):
 [1] SummarizedExperiment_1.4.0 genefilter_1.56.0         
 [3] locfit_1.5-9.1             splines_3.3.0             
 [5] lattice_0.20-34            colorspace_1.3-2          
 [7] htmltools_0.3.5            stats4_3.3.0              
 [9] base64enc_0.1-3            survival_2.40-1           
[11] XML_3.98-1.5               foreign_0.8-67            
[13] DBI_0.5-1                  BiocParallel_1.8.1        
[15] BiocGenerics_0.20.0        RColorBrewer_1.1-2        
[17] plyr_1.8.4                 stringr_1.2.0             
[19] zlibbioc_1.20.0            munsell_0.4.3             
[21] gtable_0.2.0               DESeq2_1.14.1             
[23] htmlwidgets_0.8            memoise_1.0.0             
[25] latticeExtra_0.6-28        Biobase_2.34.0            
[27] knitr_1.15.13              geneplotter_1.52.0        
[29] IRanges_2.8.1              GenomeInfoDb_1.10.3       
[31] parallel_3.3.0             AnnotationDbi_1.36.2      
[33] htmlTable_1.9              Rcpp_0.12.9.2             
[35] acepack_1.4.1              xtable_1.8-2              
[37] scales_0.4.1.9000          backports_1.0.5           
[39] checkmate_1.8.2            S4Vectors_0.12.1          
[41] Hmisc_4.0-2                annotate_1.52.1           
[43] XVector_0.14.0             gridExtra_2.2.1           
[45] ggplot2_2.2.1              digest_0.6.12             
[47] stringi_1.1.2              GenomicRanges_1.26.3      
[49] grid_3.3.0                 tools_3.3.0               
[51] bitops_1.0-6               magrittr_1.5              
[53] lazyeval_0.2.0.9000        RCurl_1.95-4.8            
[55] tibble_1.2                 RSQLite_1.1-2             
[57] Formula_1.2-1              cluster_2.0.5             
[59] Matrix_1.2-8               data.table_1.10.4         
[61] assertthat_0.1             rpart_4.1-10              
[63] nnet_7.3-12  
```
Here I am just using the example from the vignette of Glimma.
If I set the genes slot to NULL, the error is resolved. Should/can the Granges info be optional if not provided?

Happy Easter days in the very next future!
Federico

**Comment by aoles** [comment]
Hi Federico,
many thanks for spotting this and for reporting it - I will have a look soon.

Cheers,
Andrzej

**Comment by aoles** [comment]
Dear Federico,

I'm happy to inform you that this issue is resolved as of *DEFormats* version 1.15.1 (a6239f51bb7da9eebc7b53ddf2b485bfaacffdf1). Feel free to test and let me know in case there are any remaining problems.

Best,
Andrzej

**Comment by federicomarini** [comment]
Dear Andrzej,

that is very good news to hear!
I'll give it a quick run in the next iteration of development - and use it whenever I need it for analysis.

Hope to see you soon and thank you also in person in Cambridge in December!
Federico
